tests/www/v1/stock2shop/dal/channels/memory/ChannelStateTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace tests\v1\stock2shop\dal\channels\memory;
+
+use stock2shop\dal\channels\memory;
+use tests;
+
+/**
+ * Channel State Test
+ */
+class ChannelStateTest extends tests\TestCase
+{
+
+    /**
+     * Test Get Products
+     * @return void
+     */
+    public function testGetProducts()
+    {
+        memory\ChannelState::clean();
+
+        $productCount = 11;
+        $newMemoryProducts = [];
+        for ($i = 0; $i < $productCount; $i++) {
+            $newMemoryProducts[] = new memory\MemoryProduct([
+                'id' => null,
+                'product_group_id' => 'cpid1',
+                'name' => 'Product Name',
+                'price' => '5000.00',
+                'quantity' => 5,
+                'images' => [
+                    'http://aws.stock2sho..1',
+                ]
+            ]);
+        }
+
+        memory\ChannelState::updateProducts($newMemoryProducts);
+        $this->assertCount(count($newMemoryProducts), memory\ChannelState::getProducts());
+
+        memory\ChannelState::clean();
+    }
+
+    /**
+     * Test Update Images
+     * @return void
+     */
+    public function testUpdateImages()
+    {
+        memory\ChannelState::clean();
+        $memoryImageOne = new memory\MemoryImage(['id' => null, 'url' => 'http://aws.stock2sho..1', 'product_group_id' => '1']);
+        $memoryImageTwo = new memory\MemoryImage(['id' => null, 'url' => 'http://aws.stock2sho..1', 'product_group_id' => '1']);
+        memory\ChannelState::updateImages([$memoryImageOne, $memoryImageTwo]);
+        $images = memory\ChannelState::getImages();
+        foreach ($images as $imageKey => $imageItem) {
+            $this->assertTrue($imageItem instanceof memory\MemoryImage);
+            $this->assertNotNull($imageItem->id);
+            $this->assertNotNull($imageItem->url);
+            $this->assertNotNull($imageItem->product_group_id);
+        }
+        memory\ChannelState::clean();
+    }
+
+    /**
+     * Test Generate ID
+     *
+     * Evaluates the method which generates a random string ID
+     * for use with images and products on the channel.
+     *
+     * @return void
+     */
+    public function testGenerateID()
+    {
+        memory\ChannelState::clean();
```

Comment:
why do you need to clean state here?

Reply:
We are using a global variable to hold the `memory\ChannelState` object in the application. That is why I added the `clean()` method calls to each test case. But, I do understand that it is unnecessary for the `testGenerateID()` test because the outcome of the test does not depend on the state of the `memory\Channel` object.

Reply:
I have removed line 73 from `ChannelStateTest`.